### ISSUE TITLE
Mouse cursor flickering

### ISSUE THREAD

**Issue by StevePER** [gabdube/native-windows-gui#171]
In the application I have developed with native-windows-gui, the mouse cursor flickers between Normal and IBeam when positioned over a RichLabel element. It also flickers between Normal and one of the Size cursors when positioned over the window border (if the window is resizable).

In the case of RichLabel, I can workaround this by implementing a handler for WM_SETCURSOR but I do find it strange that the cursor is modified for RichLabel elements and no others.

I have been unable to workaround the flickering for the border. I suspect the flickering may be because Window specifies a cursor when instantiating WNDCLASSEXW. I've tried setting the cursor myself in my WM_SETCURSOR handler, and I've also tried calling DefWindowProcA from the handler.

**Comment by gabdube** [comment]
I;ll see what I can do. Richlabel is really just a read only rich text box, but for some reason it still sets the IBEAM cursor.

**Comment by BlackSharkfr** [comment]
I am having a similar issue.
I would like the user to be able to highlight labels with the mouse and then copy/paste elsewhere

However when I hover my mouse over the labels, the cursor briefly flashes IBeam and then return to the normal cursor.
I tried to set the global cursor manually, but the issue remains.

The documentation talks about the window class overriding the cursor unless some setting is se to NULL, but I have no idea how to do that.
https://docs.rs/native-windows-gui/1.0.11/native_windows_gui/struct.GlobalCursor.html
> impl GlobalCursor
> pub fn set(cursor: &Cursor)
> 
> Set the cursor image. If your application must set the cursor while it is in a window, make sure the class cursor for the specified window’s class is set to NULL. If the class cursor is not NULL, the system restores the class cursor each time the mouse is moved.
> 
> Arguments: • cursor: The id identifying the cursor resource

**Comment by StevePER** [comment]
Adding this code to a raw event handler prevents cursor flickering in labels. As I said previously, I haven't found a workaround for the cursor flickering when over the window border.
```rust
if msg == winapi::um::winuser::WM_SETCURSOR {
    // Ensure that the cursor doesn't flicker between the Normal cursor and the
    // IBeam cursor when over a RichLabel. The wparam parameter will be equal to
    // window_handle when over the resizable border, so we want the system to manage
    // the cursor in that case.
    if let nwg::ControlHandle::Hwnd(window_handle) = self.nwg_window.handle {
        if wparam != window_handle as usize {
            // The returned value is not used, but it needs to be a Some(_) value.
            return Some(1);
        }
    }
}
```

**Comment by gabdube** [comment]
Ok, so I wasn't able to replicate the cursor flickering on the window border. Technically, that should be handled by the non-client area of the window, which I never touched. 

That said, I removed the event override that caused flickering on the rich label . Hopefully that will fix the border flickering too.

@BlackSharkfr normal labels do not support copy pasting (that's a Windows feature, heh), but with my latest fix the rich label will work using CTRL-C. If you want your users to be able to copy using a context menu, you will have to add it yourself. For an example, check out the `show_menu` method in https://github.com/gabdube/native-windows-gui/blob/master/native-windows-gui/examples/system_tray_d.rs

 
![image](https://user-images.githubusercontent.com/5180925/116712850-43124900-a9a2-11eb-971c-89a550bc9c40.png)

**Comment by BlackSharkfr** [comment]
Thank you, this will be a nice feature.

**Comment by StevePER** [comment]
Unfortunately I still have cursor flickering after upgrading to 1.0.11, but I don't understand how this is possible given you have removed all the calls to SetCursor. I will investigate more as soon as I have a chance, which might not be for a few days.

**Comment by gabdube** [comment]
You have to point to the master branch, the fix is not yet released

**Comment by StevePER** [comment]
Sorry, I should have realised that the fix was not in 1.0.11. I've built against the master branch and there is no more cursor flickering for either the labels or the window border.

Thanks for fixing this, and also for native-windows-gui generally. It is working really well in my first serious Rust application. :)

**Comment by gabdube** [comment]
Great! Closing this then.
